src/gtc/gtir.py

```diff
@@ -105,6 +105,16 @@ def no_write_and_read_with_offset_of_same_field(
     _dtype_validation = common.assign_stmt_dtype_validation(strict=False)
 
 
+class SerialAssignStmt(common.AssignStmt[FieldAccess, Expr], Stmt):
```

Comment:
I just browsed through the PR, can you make an example for that in pseudocode or something? My first reaction is, it shouldn't exist.

Reply:
It's a hack around self assignment (i.e. `field = field[1, 0, 0]`) being forbid for parallel assignments. There have been voices strongly in favor of allowing self assignment on our team. My compromise was: if the `HorizontalMask` is a single index, then the assignment is serial rather than parallel, and in that case the `defir_to_gtir` creates one of these notes instead of the `ParAssignStmt`. This node does not have the root_validator that forbids the self-assignment.

Reply:
It is clearly unnecessary, but it means something different semantically from having to create another variable. When I started implementing regions and allowing them inside functions, some fv3 group members were skeptical because they would like to "correct" the values in the array rather than create a new array. Of course in the functional model it's not always a memory-backed variable, but here it is.

```python
def example(field: Field[float], other_field: Field[float]):
    with computation(PARALLEL), interval(...):
        other_field = some_function(field, ...)
        # Correct to account for the edge of the tile
        with horizontal(region[I[0], J[0]):
            other_field = other_field[1, 3, 0]
        with horizontal(region[I[0], J[0]+1):
            other_field = other_field[2, 3, 0]
        # ...
```

Reply:
If I read this right it would copy `other_field[1, 3, :] -> other_field[0, 0, :]` and `other_field[2, 3, :] -> other_field[0, 1, :]`, correct?

Reply:
Your question gets at the fundamental issue with horizontal regions and temporaries 🤔: Since computation is block-local, GridTools has no way to variably adjust the extents. But yes, by the GT4Py parallel model, the code should behave as is every statement is finished before the next executes.

Reply:
> If I read this right it would copy other_field[1, 3, :] -> other_field[0, 0, :] and other_field[2, 3, :] -> other_field[0, 1, :], correct?

That was almost correct, but it read `other_field[2, 4, :]` originally. I edited the example now so that your statement is correct!

Reply:
I see two methods:
1. Switch case for stencil computations in the first place, that specialize the computation on corners directly, or
2. Compute a single stencil statement everywhere, then have an if condition after that corrects the values on the corners.

It would be nice to explore (1) more, but doing so everywhere in the fv3 dycore would imply a major refactor at this point. But at the same time, currently the way FV3 is written it's performing operations on Arakawa grids, and in that system they're converting the storages from one grid to another with these corner copies. By doing so they're able to avoid using extra storages.